### PR TITLE
Make ldap integration tests independent

### PR DESCRIPTION
**Pull request by eddumelendez**
Fixes gh-5942

### CODE REVIEW

File: ldap/src/integration-test/java/org/springframework/security/ldap/AbstractLdapIntegrationTests.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2004, 2005, 2006 Acegi Technology Pty Limited
```
Comment (jzheaux) [inline ldap/src/integration-test/java/org/springframework/security/ldap/AbstractLdapIntegrationTests.java]: Thanks for watching these. Actually, for these that are marked `Acegi`, let's leave those for now.

File: ldap/src/integration-test/java/org/springframework/security/ldap/userdetails/NestedLdapAuthoritiesPopulatorTests.java [hunk 2]
```diff
@@ -73,8 +74,8 @@ public void testScalaDudeJDevelopersAuthorities() {
 		Collection<GrantedAuthority> authorities = populator.getGrantedAuthorities(ctx,
 				"scaladude");
 		assertThat(authorities).hasSize(5);
-		assertThat(Arrays.asList(javaDevelopers, scalaDevelopers,
-				circularJavaDevelopers, jDevelopers, groovyDevelopers)).isEqualTo(authorities);
+		assertThat(authorities).isEqualTo(Arrays.asList(javaDevelopers, circularJavaDevelopers,
```
[the remark this answers is not in the document]
Comment (eddumelendez) [reply ldap/src/integration-test/java/org/springframework/security/ldap/userdetails/NestedLdapAuthoritiesPopulatorTests.java]: Yes, that's the reason why these tests have changed. They have not been running before but with this new change they will

File: ldap/src/integration-test/java/org/springframework/security/ldap/AbstractLdapIntegrationTests.java [hunk 3]
```diff
@@ -15,20 +15,32 @@
...
  */
 public abstract class AbstractLdapIntegrationTests {
 	private static DefaultSpringSecurityContextSource contextSource;
 
+	private static int port;
+
 	@BeforeClass
 	public static void createContextSource() throws Exception {
-		int serverPort = ApacheDSServerIntegrationTests.getServerPort();
+		ApacheDSContainer container = new ApacheDSContainer("dc=springframework,dc=org",
```
[the remark this answers is not in the document]
Comment (eddumelendez) [reply ldap/src/integration-test/java/org/springframework/security/ldap/AbstractLdapIntegrationTests.java]: I took into account this suggestion. Thanks Josh!

**Comment by eddumelendez** [comment]
@jzheaux I have updated the PR

File: ldap/src/integration-test/java/org/springframework/security/ldap/ApacheDsContainerConfig.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+/**
+ * @author Edd� Mel�ndez
+ */
+@Configuration
+public class ApacheDsContainerConfig {
+
+	ApacheDSContainer container = new ApacheDSContainer("dc=springframework,dc=org",
+			"classpath:test-server.ldif");
+
+	public ApacheDsContainerConfig() throws Exception {
```
Comment (jzheaux) [inline ldap/src/integration-test/java/org/springframework/security/ldap/ApacheDsContainerConfig.java]: Is this constructor necessary and could you help me understand why, if so?

File: ldap/src/main/java/org/springframework/security/ldap/server/ApacheDSContainer.java [hunk 5]
```diff
@@ -186,6 +186,10 @@ public void setPort(int port) {
 		this.port = port;
 	}
 
+	public int getPort() {
```
Comment (jzheaux) [inline ldap/src/main/java/org/springframework/security/ldap/server/ApacheDSContainer.java]: Would you please put this into a separate commit and add a new ticket so that developers are alerted separately to the new feature? Thank you!

File: ldap/src/integration-test/java/org/springframework/security/ldap/ApacheDsContainerConfig.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+import javax.annotation.PreDestroy;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.ldap.core.ContextSource;
+import org.springframework.security.ldap.server.ApacheDSContainer;
+
+/**
+ * @author Edd� Mel�ndez
```
Comment (jzheaux) [inline ldap/src/integration-test/java/org/springframework/security/ldap/ApacheDsContainerConfig.java]: These look a little odd, but just for this file. Is this a GitHub glitch?

Comment (eddumelendez) [reply ldap/src/integration-test/java/org/springframework/security/ldap/ApacheDsContainerConfig.java]: that was weird. fixed

File: ldap/src/main/java/org/springframework/security/ldap/server/ApacheDSContainer.java (hunk 5 above)
Comment (eddumelendez) [reply ldap/src/main/java/org/springframework/security/ldap/server/ApacheDSContainer.java]: done

File: ldap/src/integration-test/java/org/springframework/security/ldap/ApacheDsContainerConfig.java (hunk 4 above)
Comment (eddumelendez) [reply ldap/src/integration-test/java/org/springframework/security/ldap/ApacheDsContainerConfig.java]: I made a change here so this is not needed anymore.

**Comment by eddumelendez** [comment]
@jzheaux PR updated :)

**Comment by jzheaux** [comment]
Thanks for the PR, @eddumelendez! This is now merged into `master`.
